Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -0,0 +1,316 @@
+ *
+ */
+
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/Utils/Utils.h>
+
+#include "RobotImporterWidget.h"
+#include "URDF/URDFPrefabMaker.h"
+#include "URDF/UrdfParser.h"
+#include "Utils/RobotImporterUtils.h"
+#include <QApplication>
+#include <QScreen>
+#include <QTranslator>
+
+namespace ROS2
+{
+
+    RobotImporterWidget::RobotImporterWidget(QWidget* parent)
+        : QWizard(parent)
+    {
+        m_introPage = new IntroPage(this);
+        m_fileSelectPage = new FileSelectionPage(this);
+        m_checkUrdfPage = new CheckUrdfPage(this);
+        m_assetPage = new CheckAssetPage(this);
+        m_prefabMakerPage = new PrefabMakerPage(this);
+
+        addPage(m_introPage);
+        addPage(m_fileSelectPage);
+        addPage(m_checkUrdfPage);
+        addPage(m_assetPage);
+        addPage(m_prefabMakerPage);
+
+        connect(this, &QWizard::currentIdChanged, this, &RobotImporterWidget::onCurrentIdChanged);
+        connect(m_prefabMakerPage, &QWizardPage::completeChanged, this, &RobotImporterWidget::OnUrdfCreated);
+        connect(m_prefabMakerPage, &PrefabMakerPage::onCreateButtonPressed, this, &RobotImporterWidget::onCreateButtonPressed);
+        connect(this, &RobotImporterWidget::SignalFinalizeURDFCreation, this, &RobotImporterWidget::FinalizeURDFCreation);
+        connect(
+            this,
+            &QWizard::customButtonClicked,
+            this,
+            [this](int id)
+            {
+                if (id == PrefabCreationButtonId)
+                {
+                    this->onCreateButtonPressed();
+                }
+            });
+
+        void onCreateButtonPressed();
+
+        setWindowTitle(tr("Robot Import Wizard"));
+        connect(
+            this,
+            &QDialog::finished,
+            [this](int id)
+            {
+                AZ_Printf("page", "QDialog::finished : %d", id);
+                parentWidget()->close();
+            });
+    }
+
+    void RobotImporterWidget::OnUrdfCreated()
+    {
+        // hide cancel and back buttons when last page succeed
+        if (currentPage() == m_prefabMakerPage)
+        {
+            QWizard::button(QWizard::CancelButton)->hide();
+            QWizard::button(QWizard::BackButton)->hide();
+            QWizard::button(PrefabCreationButtonId)->hide();
+        }
+    }
+
+    void RobotImporterWidget::OpenUrdf()
+    {
+        m_urdfPath = AZStd::string(m_fileSelectPage->getFileName().toUtf8().constData());
+        if (!m_urdfPath.empty())
+        {
+            AZ_Printf("Wizard", "Loading URDF file : %s", m_urdfPath.c_str());
+            m_parsedUrdf = UrdfParser::ParseFromFile(m_urdfPath);
+            QString report;
+            const auto log = UrdfParser::GetUrdfParsingLog();
+            if (m_parsedUrdf)
+            {
+                report += "# " + tr("The URDF was parsed and opened successfully") + "\n";
+                // get rid of old prefab maker
```

Comment:
No need to document, code is self explanatory
```suggestion
```

Reply:
done